src/cephadm/cephadm

```diff
@@ -486,12 +461,129 @@ class CephIscsi(object):
 ##################################
 
 
+class CustomContainer(object):
+    """Defines a custom container"""
+    daemon_type = 'container'
+
+    def __init__(self, fsid: str, daemon_id: Union[int, str],
+                 config_json: Dict, image: str) -> None:
+        self.fsid = fsid
+        self.daemon_id = daemon_id
+        self.image = image
+
+        # config-json options
+        self.entrypoint = dict_get(config_json, 'entrypoint')
+        self.uid = dict_get(config_json, 'uid', 65534)  # nobody
+        self.gid = dict_get(config_json, 'gid', 65534)  # nobody
+        self.args = dict_get(config_json, 'args', [])
+        self.volume_mounts = dict_get(config_json, 'volume_mounts', {})
+        self.container_args = dict_get(config_json, 'container_args', [])
+        self.envs = dict_get(config_json, 'envs', [])
+        self.privileged = dict_get(config_json, 'privileged', False)
+        self.ptrace = dict_get(config_json, 'ptrace', False)
+        self.bind_mounts = dict_get(config_json, 'bind_mounts', [])
+        self.ports = dict_get(config_json, 'ports', [])
+        self.dirs = dict_get(config_json, 'dirs', [])
+        self.files = dict_get(config_json, 'files', {})
```

Comment:
this provides a **huge** opportunity for us to massively simplify cephadm: If we make `CustomContainer` able to deploy all other containers as well, we can get rid of all the hardcoded daemon types in cephadm.

Reply:
I fear this is only possible if we permanently store the submitted spec when a service/daemon is applied. Currently it is not possible to `reconfig`or `redeploy` a service/daemon because the necessary information delivered by the spec are not available after it has been applied.

Reply:
> Currently it is not possible to `reconfig`or `redeploy` a service/daemon because the necessary information delivered by the spec are not available anymore at this stage.

@mgfritch  is working on this right now